docs/configuring-a-resource.md

```diff
@@ -22,126 +22,239 @@ cloud provider; sometimes it could simply be the name of resource
 (e.g. VPC id ). This is something specific to resource, and we need some input
 configuration for terrajet to appropriately generate a resource.
 
-Since Terraform already needs the same identifier to import a resource, most
+Since Terraform already needs a similar [identifier to import a resource], most
 helpful part of resource documentation is the [import section].
 
-This is [the struct that holds the External Name configuration]:
+Terrajet performs some back and forth conversions between Crossplane resource
+model and Terraform configuration. We need a custom, per resource configuration
+to adapt Crossplane `external name` from Terraform `id`.
+
+![external name configuration](images/terrajet-externalname.png)
+
+Here are [the types for the External Name configuration]:
 
 ```go
+// SetIdentifierArgumentsFn sets the name of the resource in Terraform attributes map,
+// i.e. Main HCL file.
+type SetIdentifierArgumentsFn func(base map[string]interface{}, externalName string)
+// GetExternalNameFn returns the external name extracted from the TF State.
+type GetExternalNameFn func(tfstate map[string]interface{}) (string, error)
+// GetIDFn returns the ID to be used in TF State file, i.e. "id" field in
+// terraform.tfstate.
+type GetIDFn func(ctx context.Context, externalName string, parameters map[string]interface{}, providerConfig map[string]interface{}) (string, error)
+
 // ExternalName contains all information that is necessary for naming operations,
 // such as removal of those fields from spec schema and calling Configure function
 // to fill attributes with information given in external name.
 type ExternalName struct {
-    // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
-    // map.
-    SetIdentifierArgumentFn SetIdentifierArgumentFn
-    
-    // OmittedFields are the ones you'd like to be removed from the schema since
-    // they are specified via external name. You can omit only the top level fields.
-    // No field is omitted by default.
-    OmittedFields []string
-    
-    // DisableNameInitializer allows you to specify whether the name initializer
-    // that sets external name to metadata.name if none specified should be disabled.
-    // It needs to be disabled for resources whose external name includes information
-    // more than the actual name of the resource, like subscription ID or region
-    // etc. which is unlikely to be included in metadata.name
-    DisableNameInitializer bool
+  // SetIdentifierArgumentFn sets the name of the resource in Terraform argument
+  // map. In many cases, there is a field called "name" in the HCL schema, however,
+  // there are cases like RDS DB Cluster where the name field in HCL is called
+  // "cluster_identifier". This function is the place that you can take external
+  // name and assign it to that specific key for that resource type.
```

Comment:
It would be helpful to explain, perhaps in the `ExternalName` comment block, what the distinguishing features of 'name' and 'identifier' are. When can these be the same, when should they differ. Finding that `SetIdentifierArgumentFn sets the name` confuses me because `identifier` and `name` seem to be interchangeable.

Reply:
Yeah, I agree that it is confusing with such similar words. 

May be it should be named as `SetNamingArgumentFn` to imply that it is setting the argument that gives the name of the resource?
If that sounds better, may be can consider renaming it (without breaking API of course by marking this one as deprecated and adding the other).

//cc @muvaf